test/kuzzle/query.test.js

```diff
@@ -6,6 +6,50 @@ const
   { Kuzzle } = require('../../src/Kuzzle');
 
 describe('Kuzzle query management', () => {
+  describe('#_timeoutRequest', () => {
+    beforeEach(() => {
+      const protocol = new ProtocolMock('somewhere');
+
+      kuzzle = new Kuzzle(protocol);
+    });
+
+    it('should call protocol.query', async () => {
+      await kuzzle._timeoutRequest(5000, {request: {foo: 'bar'}, options: {bar: 'baz'}});
+      should(kuzzle.protocol.query).be.calledOnce();
+      should(kuzzle.protocol.query).be.calledWith({foo: 'bar'}, {bar: 'baz'});
+    });
+
+    it('should resolve the response when a response is obtained before the configured request timeout', async () => {
+      kuzzle.protocol.query = async () => { // Simulate that we are waiting the response to a query
+        return new Promise((resolve) => {
+          setTimeout(
+            () => {
+              resolve('foo');
+            },
+            1000
```

Comment:
Can we have smaller values to not slow down the whole test suit?
(apply for subsequent test cases)